src/transformers/models/llama/modeling_llama.py

```diff
@@ -464,16 +466,12 @@ def forward(
 
         kv_seq_len = key_states.shape[-2]
         if past_key_value is not None:
-            kv_seq_len += past_key_value[0].shape[-2]
+            kv_seq_len += past_key_value.get_seq_length(self.layer_idx)
         cos, sin = self.rotary_emb(value_states, seq_len=kv_seq_len)
         query_states, key_states = apply_rotary_pos_emb(query_states, key_states, cos, sin, position_ids)
```

Comment:
In the original code, the ROPE applied to the transposed query/key states, but here query/key is not transposed, experiment result show that this will cause the shape mismatch issue.